### DISCUSSION TITLE
Diffrence between a Variable and immediate expression in debugger?

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#415] [🙋♀️TE3 Q&A] [answered]
I want to debug a value from the pivot grid, which is a simple SUM() measure. But there are filters coming from a number of places and the date is what I wanted to focus on. So I trapped the date in a variable, but it returns the maximum date in the FACT table irrespective of other filters. I need the MAX date when also filtered for the Region, which is in the rows of the pivot grid.

So I typed the same expression from the variable into the Expression box of the watch window, and *that* returns the correct date.

So help me understand why the variable and manual expression are different.

I thought it might be the expression happens at the row level (row 1 of 83 in this case) but there are actually many dates (up to 83 I suppose) all 83 rows, and only one for this region that would be the MAX date - Nov 11, 2021 as shown. 

Why is the debugger not getting that same amount from the variable, which has all of the same filters applied?

![image](https://user-images.githubusercontent.com/45298358/153088907-3740dc92-bcf2-468a-a383-617225d9f853.png)

**Comment by otykier** [comment, marked as answer]
Hi @edhans 
I just tested this, and it seems we have a bug in that the debugger ignores the outer filter context (i.e. the filter coming from the PivotGrid) when evaluating variables.

**Reply by otykier** [reply]
Hi @edhans 
This has now been fixed in [3.2.2](https://github.com/TabularEditor/TabularEditor3/releases/tag/3.2.2). Thanks for reporting!
In addition, the debugger now lets you toggle filters from the outer filter context as shown here:

![image](https://docs.tabulareditor.com/images/toggle-filters.gif)

[More information](https://docs.tabulareditor.com/te3/features/dax-debugger.html#evaluation-context).
